contracts/cw-core/src/contract.rs

```diff
@@ -541,24 +560,50 @@ pub fn query_proposal_modules(
     // the contract is still recoverable if too many items end up in
     // here.
     //
-    // Further, as the `keys` method on a map returns an iterator it
+    // Further, as the `range` method on a map returns an iterator it
     // is possible (though implementation dependent) that new keys are
     // loaded on demand as the iterator is moved. Should this be the
     // case we are only paying for what we need here.
     //
     // Even if this does lock up one can determine the existing
     // proposal modules by looking at past transactions on chain.
-    to_binary(&paginate_map_keys(
+    to_binary(&paginate_map_values(
         deps,
         &PROPOSAL_MODULES,
         start_after
             .map(|s| deps.api.addr_validate(&s))
             .transpose()?,
         limit,
-        cosmwasm_std::Order::Descending,
+        cosmwasm_std::Order::Ascending,
     )?)
 }
 
+pub fn query_active_proposal_modules(
+    deps: Deps,
+    start_after: Option<String>,
+    limit: Option<u32>,
+) -> StdResult<Binary> {
+    let values = paginate_map_values(
+        deps,
+        &PROPOSAL_MODULES,
+        start_after
+            .map(|s| deps.api.addr_validate(&s))
+            .transpose()?,
+        None,
```

Comment:
Could we add a comment to this query somewhere that this is not gas effective and needs to visit all modules? Fine for it to be this way IMO as the information is exposed elsewhere in a happy way.